Fix/FA-40-Navigation

ahmedelazab1220 opened the pull request:
<!--📌 Pull Request Template -->

<!--📝 Description -->

Refactored the navigation system by removing the BaseNavigationState abstraction. This simplifies the navigation logic and improves maintainability across screens by reducing unnecessary indirection and boilerplate.

## Changes
- Removed BaseNavigationState

- Updated screens to use simplified navigation flow

- Improved code readability and cohesion in navigation-related components

## Why
The BaseNavigationState abstraction was adding unnecessary complexity without clear benefit. This change streamlines navigation handling, making the codebase easier to understand and extend.

## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format: `type/FA-<issue-number>-TaskName`.
- [x] The branch name follows the required format: `feature/meaningful-name`, `fix/issue-description`, etc.

### 📂 Code Quality
- [x] All filenames are in `snake_case`.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with `dart format .`

### 🛠 CI/CD Validation
- [x] Linting passes with `dart analyze`.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [x] Unit tests are included for all new features and bug fixes.
- [x] `flutter test --coverage` has been run and the results reviewed.
- [x] The project builds successfully without errors.

### 📝 Additional Information
<!-- Provide any extra context or details about the changes in this PR. -->

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: `FA-40`

### 📸 Screenshots (if applicable):
<!-- Add screenshots to show the changes visually. -->

ahmedelazab1220 committed:
refactor(navigation): remove BaseNavigationState and simplify navigation handling across screens

john-safwat commented on lib/features/auth/presentation/view/widgets/login/do_not_have_account.dart:
```diff
@@ -1,20 +1,16 @@
 import '../../../../../../core/utils/routes/routes.dart';
-import '../../../view_model/login/login_cubit.dart';
 
 class DonotHaveAccount extends StatelessWidget {
```
typo error in class name it should be DoNotHaveAccount

ahmedelazab1220 committed:
fix(theme): update dialogTheme to use DialogThemeData for consistency

john-safwat commented on lib/features/auth/presentation/view/widgets/login/login_buttons.dart:
```diff
@@ -4,6 +4,7 @@ import 'package:flutter_bloc/flutter_bloc.dart';
+import '../../../../../../core/utils/routes/routes.dart';
 import '../../../view_model/login/login_cubit.dart';
 
 class LoginButtons extends StatelessWidget {
```
miss leading name for this class and no need for separating this 2 buttons in new widget instead add this buttons in the login form and defining the view model in build using context make it tide coupled with the context and the widget instead pass It to the widget constructor since it the best DI type

john-safwat commented on lib/features/auth/presentation/view/widgets/login/login_buttons.dart:
```diff
@@ -25,6 +26,7 @@ class LoginButtons extends StatelessWidget {
         ElevatedButton(
           onPressed: () {
             viewModel.doIntent(GuestRequestAction());
+            Navigator.pushReplacementNamed(context, AppRoutes.mainLayoutRoute);
```
navigation logic should be the responsibility of the view model not the view

ahmedelazab1220 replied:
this make more complexity and make state overlapping with another , i think this is not good

john-safwat commented on lib/features/auth/presentation/view/widgets/login/remember_me_and_forgot_password.dart:
```diff
@@ -48,12 +47,7 @@ class RememberMeAndForgotPassword extends StatelessWidget {
-                    type: NavigationType.push,
-                  ),
-                );
+                Navigator.pushNamed(context, AppRoutes.forgetPasswordRoute);
```
navigation logic should be the responsibility of the view model not the view

ahmedelazab1220 replied:
this make more complexity and make state overlapping with another , i think this is not good

john-safwat commented on lib/features/auth/presentation/view/widgets/register/register_form.dart:
```diff
@@ -70,6 +69,14 @@ class RegisterForm extends StatelessWidget {
                         decoration: TextDecoration.underline,
                         decorationColor: AppColors.black,
                       ),
+                      recognizer:
+                          TapGestureRecognizer()
+                            ..onTap = () {
+                              Navigator.pushNamed(
+                                context,
+                                AppRoutes.termsAndConditionsRoute,
+                              );
+                            },
                     ),
```
navigation logic should be the responsibility of the view model not the view

ahmedelazab1220 replied:
this make more complexity and make state overlapping with another , i think this is not good

john-safwat commented on lib/features/auth/presentation/view/widgets/register/register_form.dart:
```diff
@@ -99,11 +106,9 @@ class RegisterForm extends StatelessWidget {
-                                  routeName: AppRoutes.loginRoute,
-                                  type: NavigationType.pushReplacement,
-                                ),
+                              Navigator.pushReplacementNamed(
+                                context,
+                                AppRoutes.loginRoute,
```
navigation logic should be the responsibility of the view model not the view

ahmedelazab1220 replied:
this make more complexity and make state overlapping with another , i think this is not good

john-safwat commented on lib/features/edit_profile/presentation/view/widgets/edit_profile_change_password.dart:
```diff
@@ -1,12 +1,9 @@
-import '../../view_model/cubit/edit_profile_cubit.dart';
 
 class EditProfileChangePassword extends StatefulWidget {
   const EditProfileChangePassword({super.key});
```
no need to make this widget stateful widget the stateless widget will be Enough

ahmedelazab1220 replied:
this make more complexity and make state overlapping with another , i think this is not good

john-safwat commented on lib/features/edit_profile/presentation/view/widgets/edit_profile_change_password.dart:
```diff
@@ -31,11 +27,9 @@ class _EditProfileChangePasswordState extends State<EditProfileChangePassword> {
-                    routeName: AppRoutes.changePasswordRoute,
-                    type: NavigationType.pushReplacement,
-                  ),
+                Navigator.pushReplacementNamed(
+                  context,
+                  AppRoutes.changePasswordRoute,
```
navigation login is the responsibility of the viewModel

ahmedelazab1220 replied:
this make more complexity and make state overlapping with another , i think this is not good

john-safwat commented on lib/features/home/presentation/view/widgets/best_seller_item.dart:
```diff
@@ -21,16 +19,14 @@ class BestSellerItem extends StatelessWidget {
-              routeName: AppRoutes.productDetailsRoute,
-              arguments: {'productEntity': bestSellerEntity?.toProductEntity()},
-            ),
+          Navigator.pushNamed(
+            context,
+            AppRoutes.productDetailsRoute,
+            arguments: {'productEntity': bestSellerEntity?.toProductEntity()},
```
- no need for passing the arguments as map here just pass the object 
- using the key as hard coded string could lead for many logical error instead create a class for the constants and use it as variable from the class so if any typo issue happened the compiler will alert you 
- again the navigation logic should be placed in the view Model

ahmedelazab1220 replied:
- i prefer pass as map to clear what is type object i pass it this make it clear  
- this make more complexity and make state overlapping with another , i think this is not good

john-safwat reviewed (commented):
- there is a miss understanding of the concept of composition pattern the items related together should be in the same widget not diving the widgets for the sake of dividing widgets
- building shared widget is not for the repeated widgets only it can be used in may ways for example in Home Screen the horizontal lists can be shared widget the you pass the widget type as template parameters and items list and any needed call back so you can reuse the same widget for different type of lists

ahmedelazab1220 committed:
chore(dependencies): update skeletonizer to version 2.0.1

ahmedelazab1220 committed:
feat(localization): add success messages for profile edits and uploads, and update translations
refactor(constants): centralize constant values for better maintainability
fix(naming): correct class name from DonotHaveAccount to DoNotHaveAccount
chore(cleanup): remove generated assets file
